fragment.go

```diff
@@ -118,6 +118,8 @@ type fragment struct {
 
 	// Stats reporting.
 	maxRowID uint64
+	minRowID uint64
+	hasRowID bool
```

Comment:
I don't see much benefit to caching `hasRowID` and `minRowID` on the fragment. They shouldn't be very expensive to look up (without a filter), and this brings up the opportunity for cache invalidation bugs.

Reply:
Moved `minRowID` to a function.